src/System.Net.Security/src/System/Net/SecureChannel.cs

```diff
@@ -1059,16 +1060,12 @@ internal SecurityStatusPal Encrypt(byte[] buffer, int offset, int size, ref byte
                 resultSize = 0;
 
                 int bufferSizeNeeded = checked(size + _headerSize + _trailerSize);
-                if (output != null && bufferSizeNeeded <= output.Length)
-                {
-                    writeBuffer = output;
-                }
-                else
+
```

Comment:
Yeah, I left this here just so the potential exception flow wouldn't change. But I guess it's silly now that I set the vales to 0 on Unix.